pkg/provider/azure_standard.go

```diff
@@ -261,44 +263,50 @@ func isInternalLoadBalancer(lb *network.LoadBalancer) bool {
 // This means:
 // clusters moving from IPv4 to dualstack will require no changes
 // clusters moving from IPv6 to dualstack will require no changes as the IPv4 backend pool will created with <clusterName>
-func getBackendPoolName(clusterName string, service *v1.Service) string {
-	IPv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
-	if IPv6 {
-		return fmt.Sprintf("%v-IPv6", clusterName)
+func getBackendPoolName(clusterName string, isIPv6 bool) string {
+	if isIPv6 {
+		return fmt.Sprintf("%s-%s", clusterName, v6Suffix)
 	}
 
 	return clusterName
 }
 
-func (az *Cloud) getLoadBalancerRuleName(service *v1.Service, protocol v1.Protocol, port int32) string {
+// ifBackendPoolIPv6 checks if a backend pool is of IPv6 according to name/ID.
+func isBackendPoolIPv6(name string) bool {
+	return strings.HasSuffix(name, "-IPv6")
```

Comment:
Use 'v6Suffix'?

Reply:
Updated to `fmt.Sprintf("-%s", v6Suffix)`